esmvalcore/preprocessor/_io.py

```diff
@@ -347,7 +347,34 @@ def _get_debug_filename(filename, step):
 
 
 def cleanup(files, remove=None):
-    """Clean up after running the preprocessor."""
+    """Clean up after running the preprocessor.
+
+    Warning
+    -------
+    .. deprecated:: 2.8.0
+        This function has been deprecated in ESMValCore version 2.8.0 and is
+        scheduled for removal in version 2.10.0.
```

Comment:
```suggestion
        This function is no longer used and has been deprecated since
        ESMValCore version 2.8.0. It is scheduled for removal in version 2.10.0.
```